Source/ThirdParty/libwebrtc/CMakeLists.txt

```
@@ -2154,14 +2154,23 @@ else ()
 
         Source/webrtc/modules/audio_device/linux/alsasymboltable_linux.cc
         Source/webrtc/modules/audio_device/linux/audio_device_alsa_linux.cc
-        Source/webrtc/modules/audio_device/linux/audio_device_pulse_linux.cc
         Source/webrtc/modules/audio_device/linux/audio_mixer_manager_alsa_linux.cc
-        Source/webrtc/modules/audio_device/linux/audio_mixer_manager_pulse_linux.cc
         Source/webrtc/modules/audio_device/linux/latebindingsymboltable_linux.cc
-        Source/webrtc/modules/audio_device/linux/pulseaudiosymboltable_linux.cc
 
         Source/webrtc/modules/video_coding/codecs/h264/h264.cc
     )
+
+    find_package(LibPulse)
+    if (NOT LIBPULSE_FOUND)
+        message(STATUS "libpulse is not found, not building support.")
+    else()
+        list(APPEND webrtc_SOURCES
+            Source/webrtc/modules/audio_device/linux/audio_device_pulse_linux.cc
+            Source/webrtc/modules/audio_device/linux/audio_mixer_manager_pulse_linux.cc
+            Source/webrtc/modules/audio_device/linux/pulseaudiosymboltable_linux.cc
+        )
```

Don't we need to add to the include path here and possibly the compile options (if any) returned by `pkg-config`? Something like:

```suggestion
        )
        target_include_directories(webrtc PRIVATE ${LibPulse_INCLUDE_DIRS})
        target_compile_options(webrtc PRIVATE ${LibPulse_COMPILE_OPTIONS})
```

Otherwise if the PulseAudio headers are in a non-standard location, or if they need some particular compiler flag or preprocessor definition.

Alternatively, the above could be set only for the three source files:

```suggestion
        )
        foreach (source audio_device_pulse_linux.cc audio_mixer_manager_pulse_linux.cc pulseaudiosymboltable_linux.cc)
            set_property(
                SOURCE Source/webrtc/modules/audio_device/linux/${source}
                APPEND PROPERTY INCLUDE_DIRECTORIES ${LibPulse_INCLUDE_DIRS}
            )
            set_property(
                SOURCE Source/webrtc/modules/audio_device/linux/${source}
                APPEND PROPERTY COMPILE_OPTIONS ${LibPulse_COMPILE_OPTIONS}
            )
        endforeach ()
```